File: CodeMaze-master/src/counting_bits/solution3.swift
class Solution {
	// time complexity: O(n), space complexity: O(n)
    func countBits(_ num: Int) -> [Int] {
        var res = [Int](repeating: 0, count: num+1)
        
        if num > 0 {
        
            for i in 1 ... num {
                res[i] = res[i&(i-1)] + 1
            }
        }
        
        return res
    }
}

File: CodeMaze-master/src/counting_bits/solution3.cpp
class Solution {
public:
	// time complexity: O(n), space complexity: O(n)
    vector<int> countBits(int num) {
        vector<int> results = vector<int>(num+1, 0);
        if (num > 0) {
            for (int i = 1; i <= num; ++i) {
                results[i] = results[i & (i-1)] + 1;
            }
        }
        return results;
    }
};